dashboard/src/main/home/infrastructure/ExpandedInfra.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useContext, useEffect, useState } from "react";
+import styled from "styled-components";
+import DynamicLink from "components/DynamicLink";
+import Loading from "components/Loading";
+import { Context } from "shared/Context";
+import api from "shared/api";
+import { integrationList } from "shared/common";
+import yaml from "js-yaml";
+
+import DeployList from "./components/DeployList";
+import InfraResourceList from "./components/InfraResourceList";
+import PorterFormWrapper from "components/porter-form/PorterFormWrapper";
+import { readableDate } from "shared/string_utils";
+import Placeholder from "components/Placeholder";
+import Header from "components/expanded-object/Header";
+import { Infrastructure, KindMap } from "shared/types";
+
+type Props = {
+  infra_id: number;
+};
+
+type InfraTabOptions = "deploys" | "resources" | "settings";
+
+const ExpandedInfra: React.FunctionComponent<Props> = ({ infra_id }) => {
+  const [infra, setInfra] = useState<Infrastructure>(null);
+  const [hasError, setHasError] = useState(false);
+
+  const { currentProject, setCurrentError } = useContext(Context);
+
+  useEffect(() => {
+    if (!currentProject) {
+      return;
+    }
+
+    let isSubscribed = true;
+
+    api
+      .getInfraByID(
+        "<token>",
+        {},
+        {
+          project_id: currentProject.id,
+          infra_id: infra_id,
+        }
+      )
+      .then(({ data }) => {
+        if (!isSubscribed) {
+          return;
+        }
+
+        setInfra(data);
+      })
+      .catch((err) => {
+        console.error(err);
+        setHasError(true);
+        setCurrentError(err.response?.data?.error);
+      });
+  }, [currentProject, infra_id]);
+
+  if (hasError) {
+    return <Placeholder>Error loading infra</Placeholder>;
+  }
+
+  if (!infra) {
+    return <Loading />;
+  }
+
+  const renderTabContents = (newTab: InfraTabOptions) => {
+    switch (newTab) {
+      case "deploys":
+        return <DeployList infra_id={infra_id} />;
+      case "resources":
+        return <InfraResourceList infra_id={infra_id} />;
+      case "settings":
+        return <div>Settings</div>;
+    }
+  };
+
+  const formData = yaml.load(initYaml);
+
+  return (
+    <StyledExpandedInfra>
+      <Header
+        last_updated={readableDate(infra.latest_operation?.last_updated)}
+        back_link={"/infrastructure"}
+        name={integrationList[infra.kind].label}
+        icon={integrationList[infra.kind].icon}
```

Comment:
Just in case, can we add a null casing here (as well as other places using the integrationList)? Wouldn't be the first time where the component cannot render because of the index being undefined/null. 

```suggestion
        name={integrationList[infra.kind]?.label}
        icon={integrationList[infra.kind]?.icon}
```